### ISSUE TITLE
Prevent the removal of the last admin

### ISSUE THREAD

**Issue by hardillb** [flowforge/flowforge#952]
### Description

Update DELETE `/api/v1/users/:id` to fail if trying to remove the last user with the admin flag set.

**Comment by Steve-Mcl** [comment]
@hardillb I tried this on latest running git and this is the following scenario...

### An admin cannot be deleted due to this ([User.js L62-L64](https://github.com/flowforge/flowforge/blob/75aea3d53055fe3e4ba1bec30515ce3982c56923/forge/db/models/User.js#L62-L64))... 
```
                if (user.admin) {
                    throw new Error('Cannot delete Admin user')
                }
```

So the job is potentially to prevent the removal of admin flag if user is last admin?

**Comment by hardillb** [comment]
OK, that makes sense
You can't delete an admin user, you have to demote them first (which you have to be an admin to do)
And if you demote yourself you can't then delete yourself as we don't allow users to delete their own accounts at the moment.

So I'll say we can close this for now. If we let users remove their own accounts then we will need to re-examine.

**Comment by hardillb** [comment]
Sorry, miss read the last bit.

Yes, don't remove admin flag from yourself if you are the last admin

**Comment by hardillb** [comment]
`app.db.models.scope('admins')` is probably going to be your friend here

**Comment by Steve-Mcl** [comment]
NP, I'll do that. 

There are a couple of places I can do that - in the API or in the model (beforeUpdate) 

Which would you chose?

**Comment by hardillb** [comment]
In the model in a similar fashion to the last test makes most sense I think.

**Comment by Steve-Mcl** [comment]
So I have this pretty much working but the `ff-dialog` auto closes meaning I cannot sensibly provide feedback.

I have [raised this](https://github.com/flowforge/flowforge/issues/966) as a bug (since it affects other server-side validation issues on this form)

Related: https://github.com/flowforge/forge-ui-components/issues/48

---

For reference / traceability... 

### Implementation order...
1. implement https://github.com/flowforge/forge-ui-components/issues/48
2. fix https://github.com/flowforge/flowforge/issues/966
3. implement this PR

NOTE: implementing steps 1 and 3 will close also fix #966 (step 2)
